src/systemcmds/mtd/mtd.cpp

```diff
@@ -471,7 +505,18 @@ static ssize_t mtd_get_partition_size(const mtd_instance_s &instance)
 		return 0;
 	}
 
-	return partsize;
+	unsigned partn = 0;
+
+	for (unsigned n = 0; n < instance.n_partitions_current; n++) {
+		if (instance.partition_names[n] != nullptr &&
+		    partname != nullptr &&
+		    strcmp(instance.partition_names[n], partname) == 0) {
+			partn = n;
+			break;
+		}
+	}
+
+	return instance.partition_block_counts[partn] * erasesize;
```

Comment:
Also here should be:
`return instance.partition_block_counts[partn] * blocksize;`

To return correct size of partition as before
https://github.com/PX4/PX4-Autopilot/blob/615333badb289aa77285bcfad071cc6a70b13632/src/systemcmds/mtd/mtd.cpp#L488

Otherwise readtest get failed because expected != correctly read partition size.

Reply:
@lukegluke - Thank you for the review! 
This is fixed in https://github.com/PX4/PX4-Autopilot/pull/16212 and devices are blkpererase = 1 so that erasesize == blocksize for PX4 device used.